src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,241 @@
+def _is_py_class_model(model_path):
+    return model_path[-3:] == ".py"
+
+
+def _import_module_from_path(module_name, file_path):
+    spec = importlib.util.spec_from_file_location(module_name, file_path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+def _get_model_class_from_module(module):
+    names = dir(module)
+    for name in names:
+        attr = getattr(module, name)
+        try:
+            if issubclass(attr, torch.nn.Module):
+                return attr
+        except TypeError:
+            # attr may not be a class
+            pass
+    raise pb_utils.TritonModelException("Cannot find a subclass of torch.nn.Module")
+
+
+def _parse_io_config(io_config):
+    io = []
+    for conf in io_config:
+        io.append({"name": conf["name"]})
+    return io
+
+
+def _get_device_name(model_instance_kind, model_instance_device_id):
+    if model_instance_kind == "GPU":
+        return "cuda:" + model_instance_device_id
+    return "cpu"
+
+
+def _enable_torch_compile(config):
+    if int(torch.__version__.split(".")[0]) < 2:
+        # torch.compile is supported starting from PyTorch 2.0
+        return False
+    if "DISABLE_TORCH_COMPILE" in config["parameters"]:
+        val = config["parameters"]["DISABLE_TORCH_COMPILE"]["string_value"].upper()
+        if val == "YES" or val == "TRUE" or val == "1":
+            return False
+    return True
+
+
+class TritonPythonModel:
+    """Your Python model must use the same class name. Every Python model
+    that is created must have "TritonPythonModel" as the class name.
+    """
+
+    def initialize(self, args):
+        """`initialize` is called only once when the model is being loaded.
+        Implementing `initialize` function is optional. This function allows
+        the model to initialize any state associated with this model.
+
+        Parameters
+        ----------
+        args : dict
+          Both keys and values are strings. The dictionary keys and values are:
+          * model_config: A JSON string containing the model configuration
+          * model_instance_kind: A string containing model instance kind
+          * model_instance_device_id: A string containing model instance device ID
+          * model_repository: Model repository path
+          * model_version: Model version
+          * model_name: Model name
+        """
+        self._model_name = args["model_name"]
+        self._logger = pb_utils.Logger
+        self._logger.log_info(
+            "Initializing model instance for '" + self._model_name + "'"
+        )
+
+        self._model_config = json.loads(args["model_config"])
+        self._inputs = _parse_io_config(self._model_config["input"])
+        self._outputs = _parse_io_config(self._model_config["output"])
+
+        kind = args["model_instance_kind"]
+        device_id = args["model_instance_device_id"]
+        self._device = torch.device(_get_device_name(kind, device_id))
```

Comment:
What about using KIND_MODEL? We should ideally not set explicit devices when using KIND_MODEL.

Reply:
Yes, `KIND_MODEL` is now supported. If specified, the torch object will be left at its default device.